builtin/logical/pki/path_config_ca.go

```diff
@@ -48,24 +470,57 @@ func (b *backend) pathCAWrite(
 		parsedBundle.CertificateBytes = parsedBundle.IssuingCABytes
 	}
 
-	// TODO?: CRLs can only be generated with RSA keys right now, in the
-	// Go standard library. The plubming is here to support non-RSA keys
-	// if the library gets support
+	cb := &certutil.CertBundle{}
+	entry, err := req.Storage.Get("config/ca_bundle")
+	if err != nil {
+		return nil, err
+	}
+	if entry != nil {
+		err = entry.DecodeJSON(cb)
+		if err != nil {
+			return nil, err
+		}
+		// If we have a stored private key and did not get one, attempt to
+		// correlate the two -- this could be due to a CSR being signed
+		// for a generated CA cert and the resulting cert now being uploaded
+		if len(cb.PrivateKey) != 0 &&
+			cb.PrivateKeyType != "" &&
+			parsedBundle.PrivateKeyType == certutil.UnknownPrivateKey &&
+			(parsedBundle.PrivateKeyBytes == nil || len(parsedBundle.PrivateKeyBytes) == 0) {
+			parsedCB, err := cb.ToParsedCertBundle()
+			if err != nil {
+				return nil, err
+			}
+			if parsedCB.PrivateKey == nil {
+				return nil, fmt.Errorf("Encountered nil private key from saved key")
+			}
+			// If true, the stored private key corresponds to the cert's
+			// public key, so fill it in
+			//panic(fmt.Sprintf("\nparsedCB.PrivateKey.Public().: %#v\nparsedBundle.Certificate.PublicKey"))
+			if reflect.DeepEqual(parsedCB.PrivateKey.Public(), parsedBundle.Certificate.PublicKey) {
```

Comment:
OK -- I will attempt type conversion to both rsa and ec, then compare those parameters.